security/pkg/pki/ca/controller/secret.go

```diff
@@ -258,6 +262,11 @@ func (sc *SecretController) scrtDeleted(obj interface{}) {
 
 func (sc *SecretController) generateKeyAndCert(saName string, saNamespace string) ([]byte, []byte, error) {
 	id := fmt.Sprintf("%s://cluster.local/ns/%s/sa/%s", util.URIScheme, saNamespace, saName)
+	if sc.webhookServiceMap != nil {
+		if svc, ok := sc.webhookServiceMap[saName]; ok {
```

Comment:
Only matching the service account is not enough. We need to match the namespace as well.